electricitymap/contrib/parsers/OPENNEM.py

```diff
@@ -533,11 +500,94 @@ def _fetch_au_nsw_au_vic_exchange(
     return datetimes_and_netflows
 
 
+def _build_network_url(
+    path: str,
+    network_code: str,
+    metrics: list[str],
+    target_datetime: datetime,
+    network_region: str,
+) -> str:
+    target_datetime_local = target_datetime.astimezone(ZoneInfo("Australia/Sydney"))
+    target_datetime_naive = target_datetime_local.replace(tzinfo=None)
+
+    date_start = (target_datetime_naive - REFETCH_FREQUENCY).strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+    date_end = target_datetime_naive.strftime("%Y-%m-%dT%H:%M:%S")
+
+    params = [("metrics", m) for m in metrics]
+    params += [
+        ("date_start", date_start),
+        ("date_end", date_end),
+        ("network_region", network_region),
+    ]
+
+    query = urlencode(params)
+    base = f"https://api.openelectricity.org.au/v4/{path}/network/{network_code}"
+
+    return f"{base}?{query}"
+
+
+def _fetch_network_datasets(
+    zone_key: str,
+    session: Session,
+    dataset_type: str,
+    target_datetime: datetime,
+    metrics: list[str],
+) -> list[dict[str, Any]]:
+    network_region = ZONE_KEY_TO_REGION.get(zone_key)
+    network_code = ZONE_KEY_TO_NETWORK.get(zone_key)
+
+    if not network_region:
+        raise ParserException(
+            parser="OPENNEM",
+            message=f"Invalid zone_key {zone_key}, valid keys are {list(ZONE_KEY_TO_REGION.keys())}",
+            zone_key=zone_key,
+        )
+
+    url = _build_network_url(
+        path=dataset_type,
+        network_code=network_code,
+        metrics=metrics,
+        target_datetime=target_datetime,
+        network_region=network_region,
+    )
+
+    token = get_token("OPEN_ELECTRICITY_API_KEY")
+    headers = {
+        "Authorization": f"Bearer {token}",
+    }
+
+    response = session.get(url, headers=headers)
+    response.raise_for_status()
+
+    return response.json()["data"]
+
+
+def _build_price_list(datasets, zone_key: ZoneKey, logger: Logger) -> PriceList:
+    price_list = PriceList(logger=logger)
+    for dataset in datasets:
+        if dataset["metric"] != "price":
+            continue
+        for result in dataset["results"]:
+            for ts, price in result["data"]:
+                dt = datetime.fromisoformat(ts).astimezone(timezone.utc)
```

Comment:
```suggestion
                dt = datetime.fromisoformat(ts)
```
There is no need for this. We just need a datetime with a timezone.